src/3d/qgs3dutils.cpp

```diff
@@ -558,14 +558,14 @@ static inline uint outcode( QVector4D v )
  * qt3d /src/threed/painting/qglpainter.cpp
  * bool QGLPainter::isCullable(const QBox3D& box) const
  */
-bool Qgs3DUtils::isCullable( const QgsAABB &bbox, const QMatrix4x4 &viewProjectionMatrix )
+bool Qgs3DUtils::isCullable( const QgsAABB &bbox, const QMatrix4x4 &viewProjectionMatrix, float verticalOffset )
 {
   uint out = 0xff;
 
   for ( int i = 0; i < 8; ++i )
   {
     const QVector4D p( ( ( i >> 0 ) & 1 ) ? bbox.xMin : bbox.xMax,
-                       ( ( i >> 1 ) & 1 ) ? bbox.yMin : bbox.yMax,
+                       ( ( i >> 1 ) & 1 ) ? bbox.yMin + verticalOffset : bbox.yMax + verticalOffset,
```

Comment:
This looks odd to me -- I think the passed bbox should just be shifted prior to passing to this function

Reply:
Done.